app/actions/pricesActions.js

```diff
@@ -2,32 +2,43 @@
 import axios from 'axios'
 import { createActions } from 'spunky'
 
-import { DEFAULT_CURRENCY_CODE } from '../core/constants'
+import { getDefaultTokens } from '../core/nep5'
+import { DEFAULT_CURRENCY_CODE, ASSETS } from '../core/constants'
 
 type Props = {
   currency?: string
 }
 
-function mapPrices(tickers, currency) {
+function mapPrices(pricingData: Array<any>, currency) {
   const mapping = {}
-
-  tickers.forEach(ticker => {
-    mapping[ticker.symbol] = parseFloat(
-      ticker[`price_${currency.toLowerCase()}`]
-    )
+  const upperCasedCurrency = currency.toUpperCase()
+  pricingData.forEach((price: { currency: { upperCasedCurrency: Object } }) => {
+    const priceInSelectedCurrency = price[upperCasedCurrency]
+    if (price && priceInSelectedCurrency) {
+      mapping[priceInSelectedCurrency.FROMSYMBOL] = parseFloat(
+        priceInSelectedCurrency.PRICE
+      )
+    }
   })
 
   return mapping
 }
 
-function getPrices(currency) {
-  const url = `https://api.coinmarketcap.com/v1/ticker/?limit=0&convert=${currency.toLowerCase()}`
+async function getPrices(currency) {
+  const tokens = await getDefaultTokens()
+  const joinedTokens = tokens
+    .map(token => token.symbol)
+    .concat([ASSETS.NEO, ASSETS.GAS])
+    .join(',')
+  const url = `https://min-api.cryptocompare.com/data/pricemultifull?fsyms=${joinedTokens}&tsyms=${currency.toUpperCase()}`
 
-  return axios.get(url).then(response => {
-    const { data } = response
-    if (data.error) throw new Error(data.error)
-    return mapPrices(data, currency)
-  })
+  const priceDataResponse = await axios
```

Comment:
could you wrap this inside a try/catch (also put `pricingArray` inside it), that will help catching the errors that might arise if the response changes, etc.

Reply:
we could also use joi to validate response data